runtime/permissions/lib.rs

```diff
@@ -689,11 +691,39 @@ impl Descriptor for WriteDescriptor {
 }
 
 #[derive(Clone, Eq, PartialEq, Hash, Debug)]
-pub struct NetDescriptor(pub FQDN, pub Option<u16>);
+pub struct NetDescriptor(pub Host, pub Option<u16>);
 
 impl NetDescriptor {
-  fn new<T: AsRef<str>>(host: &&(T, Option<u16>)) -> Self {
-    NetDescriptor(fqdn!(host.0.as_ref()), host.1)
+  pub fn from_url(url: &Url) -> Result<Self, AnyError> {
+    let hostname = url.host_str().ok_or(ParseError::EmptyHost)?.to_string();
+    let host = Host::from_str(&hostname, &hostname.clone())?;
+    let port = url.port_or_known_default();
+    Ok(NetDescriptor(host, port))
+  }
+
+  /// Parses a string representation of a network descriptor.
+  ///
+  /// # Errors
+  ///
+  /// Returns an `AnyError` if the input string cannot be parsed.
+  pub fn from_str(s: &str) -> Result<Self, AnyError> {
```

Comment:
Again, I think `clippy` will complain about this one. Is this one meant to parse the CLI args? If so I suggest to rename it to `from_cli_arg()`

Reply:
it's for extract the net descriptor from str 
 
 when do --allow-net=https://github.com:443/
 
 it will extract the  the NetDescriptor(hostname:github.com ,port:443)